### DISCUSSION TITLE
How  can I add custom filter for the tempo traces  visualization.

### DISCUSSION THREAD

**Discussion by nikhil-ht** [grafana/tempo#3662] [General]
I have attached the opentelemetry agent to the API services and used the  below env 

```
ENV OTEL_TRACES_EXPORTER=otlp
ENV OTEL_METRICS_EXPORTER=none
ENV OTEL_EXPORTER_OTLP_ENDPOINT=$OTEL_EXPORTER_OTLP_ENDPOINT
ENV OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
ENV OTEL_RESOURCE_ATTRIBUTES=service.name=example-api
ENV CATALINA_OPTS="-javaagent:/opentelemetry-javaagent-all.jar" 

```

I want to add a custom filter in Grafana Tempo. This filter will categorize the services based on the environment, such as QA, DEV, Stage, etc.

![image](https://github.com/grafana/tempo/assets/92708869/37b724e6-a4ae-4fd3-8ec3-13ca9ba8c971)

**Comment by joe-elliott** [comment]
You appear to be using a quite old version of Grafana. I would recommend upgrading. The latest versions of Grafana provide a much better search experience. In the current UI you can easily filter on any span or resource attribute:

![image](https://github.com/grafana/tempo/assets/2272392/e6d7777d-1d63-4555-b146-71fdf94ea432)
